### PR TITLE
Run unit tests via concourse pipeline

### PR DESCRIPTION
**Pull request by mvach**

### CODE REVIEW

**Review by ramonskie — APPROVED** [review]
lgtm

File: ci/scripts/run-unit-tests.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env sh
+set -euo pipefail
+
+# log current go version
+go version
+
+# install gcc
+apk --no-progress add build-base
```
Comment (rkoster) [inline ci/scripts/run-unit-tests.sh:8]: Why is this line needed?

Comment (mvach) [reply ci/scripts/run-unit-tests.sh:8]: build-base contains gcc

Comment (mvach) [reply ci/scripts/run-unit-tests.sh:8]: Hi @rkoster,

This is to fix:
```
...
go: downloading golang.org/x/net v0.8.0
go: downloading golang.org/x/sys v0.6.0
go: downloading gopkg.in/yaml.v3 v3.0.1
go: downloading golang.org/x/text v0.8.0
# runtime/cgo
cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in $PATH

Failed to compile config:

# runtime/cgo
cgo: C compiler "gcc" not found: exec: "gcc": executable file not found in $PATH
```

Comment (rkoster) [reply ci/scripts/run-unit-tests.sh:8]: Would compiling with `CGO_ENABLED=0` also work?

Comment (mvach) [reply ci/scripts/run-unit-tests.sh]: Ahh nice hint, I updated the PR accordingly.
THX!!
